usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js

```diff
@@ -7,30 +7,45 @@
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 
 (function() {
-    var ORIGIN = {x:0, y:0, z: 0};
-    var RADIUS = 100000;
     
     var _this;
-    var roomName;
-    var occupantTextEntityId;
 
-    var MeetingZone = function() {
-        _this = this;        
+    var MeetingZone = function() {  
+        _this = this;     
     };
 
     MeetingZone.prototype = {
-        preload: function(entityId) {
-            _this.entityId = entityId;
-            roomName = Entities.getEntityProperties(_this.entityId, ['name']).name;
-            occupantTextEntityId = Entities.findEntitiesByName(roomName + '_OCCUPANTS', ORIGIN, RADIUS)[0];
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            _this.userData = Entities.getEntityProperties(_this.entityID, ['userData']).userData;
+            AvatarManager.avatarRemovedEvent.connect(_this.leaveDomain);
+            if (_this.userData.length !== 0) {
+                try {
+                    _this.userData = JSON.parse(_this.userData);
+                } catch (e) {
+                    console.log(e, "Could not parse userData");
+                    return;
+                }
+                _this.occupantsListID = _this.userData.occupantsListID;
+            } else {
+                console.log("No userData found");
```

Comment:
in this case, you want to `return`, right? (not that there's anything below this, but for security for adding code later)